client/my-sites/stats/site.jsx

```diff
@@ -148,9 +149,15 @@ module.exports = React.createClass( {
 						queryDate={ queryDate }
 						period={ this.props.period }
 						chartTab={ this.state.chartTab } />
-					<DatePicker
-						period={ period }
-						date={ this.state.chartDate } />
+					<StatsPeriodNavigation
+						date={ this.props.date }
+						period={ this.props.period.period }
+						url={ `/stats/${ this.props.period.period }/${ site.slug }` }
```

Comment:
> Thought, we already store selectedSiteId and editedPostId

That is why I thought of storing the `period` in props too.  Just a random idea.